### DISCUSSION TITLE
When to use `execute_batch`?

### DISCUSSION THREAD

**Discussion by zjp-CN** [rusqlite/rusqlite#1236] [Questions] [answered]
`execute` finishes one sql query at a time. So will `execute_batch` be more efficient if multiple sql queries are required?

But after I glance at the source code, I find `execute_batch` actually break down queries and process one at a time.
https://github.com/rusqlite/rusqlite/blob/a37489ec93c8ec940f61dd76de34d8936336dca7/src/lib.rs#L510-L521

My question is whether sqlite receives multiple requries that are passed to `execute_batch`?

**Comment by gwenn** [comment, marked as answer]
`execute_batch` is like [sqlite3_exec](https://github.com/sqlite/sqlite/blob/37d4ec86bfa78c31732132b7729b8ce0e47da891/src/legacy.c#L48) but without callback.
 SQLite [doc](http://sqlite.org/c3ref/exec.html) says:
> The sqlite3_exec() interface is a convenience wrapper around [sqlite3_prepare_v2()](http://sqlite.org/c3ref/prepare.html), [sqlite3_step()](http://sqlite.org/c3ref/step.html), and [sqlite3_finalize()](http://sqlite.org/c3ref/finalize.html), that allows an application to run multiple statements of SQL without having to use a lot of C code.

So this doesn't mean more efficient, just more convenient.
And both `execute_batch` and `sqlite3_exec` prepare and execute multiple queries.
